foreman/data_refinery_foreman/foreman/management/commands/create_missing_processor_jobs.py

```diff
@@ -0,0 +1,193 @@
+            original_file.save()
+        else:
+            processor_job = ProcessorJob()
+            processor_job.pipeline_applied = pipeline_to_apply.value
+            processor_job.ram_amount = determine_ram_amount(sample_object, processor_job)
+            processor_job.volume_index = volume_index
+            processor_job.save()
+
+            assoc = ProcessorJobOriginalFileAssociation()
+            assoc.original_file = original_file
+            assoc.processor_job = processor_job
+            assoc.save()
+
+            logger.debug("Queuing processor job.",
+                         processor_job=processor_job.id,
+                         original_file=original_file.id)
+
+            send_job(pipeline_to_apply, processor_job)
+
+
+def create_processor_job_for_original_files(original_files: List[OriginalFile],
+                                            volume_index: int):
+    """
+    Create a processor job and queue a processor task for sample related to an experiment.
+    """
+    # If there's no original files then we've created all the jobs we need to!
+    if len(original_files) == 0:
+        return
+    # For anything that has raw data there should only be one Sample per OriginalFile
+    sample_object = original_files[0].samples.first()
+    pipeline_to_apply = determine_processor_pipeline(sample_object, original_files[0])
+    if pipeline_to_apply == ProcessorPipeline.NONE:
+        logger.info("No valid processor pipeline found to apply to sample.",
+                    sample=sample_object.id,
+                    original_file=original_files[0].id)
+        for original_file in original_files:
+            original_file.delete_local_file()
+            original_file.is_downloaded = False
+            original_file.save()
+    else:
+        processor_job = ProcessorJob()
+        processor_job.pipeline_applied = pipeline_to_apply.value
+        processor_job.ram_amount = determine_ram_amount(sample_object, processor_job)
+        processor_job.volume_index = volume_index
+        processor_job.save()
+        for original_file in original_files:
+            assoc = ProcessorJobOriginalFileAssociation()
+            assoc.original_file = original_file
+            assoc.processor_job = processor_job
+            assoc.save()
+        logger.debug("Queuing processor job.",
+                     processor_job=processor_job.id)
+        send_job(pipeline_to_apply, processor_job)
+
+
+def find_volume_index_for_dl_job(job: DownloaderJob) -> int:
+    pjs = ProcessorJob.objects.filter(worker_id=job.worker_id)
+
+    if pjs:
+        return pjs.first().volume_index
+    else:
+        logger.warn(
+            "Could not determine what volume index a downloader job was run on",
+            downloader_job=job.id
+        )
+        raise ValueError()
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        logger.info(DownloaderJob.objects.filter(success='t').count())
```

Comment:
Might want to use this queryset, and I don't know how Django will handle this. It should be success=True idiomatically

Reply:
This was for development. I had already removed it but I forgot to commit that change.